tests/detection.rs

```diff
@@ -0,0 +1,68 @@
+extern crate libtest_mimic;
+
+use libtest_mimic::{Arguments, Failed, Trial};
+use std::{
+    fs::read_dir,
+    process::{Command, ExitCode, Stdio},
+};
+
+fn main() -> ExitCode {
+    let args = Arguments::from_args();
+
+    let mut test_cases: Vec<String> = Vec::new();
+
+    // collect test cases
+    for file in read_dir("tests/detection/examples").unwrap() {
+        let file = file.unwrap();
+        // println!("{:?}", file);
+        assert!(file.metadata().unwrap().is_file());
+
+        test_cases.push(
+            file.file_name()
+                .into_string()
+                .unwrap()
+                .strip_suffix(".rs")
+                .unwrap()
+                .to_string(),
+        );
+    }
+
+    #[expect(unexpected_cfgs)]
+    let ignored = !cfg!(rtsan_supported);
+
+    if ignored {
+        println!("WARNING: RTSAN not supported. Skipping detection tests");
```

Comment:
I currently this is not skipping anything?

```suggestion
        println!("WARNING: RTSAN not supported. Skipping detection tests");
        return ExitCode::SUCCESS;
```

Reply:
Ah sorry missed the ignored flag at the end, but it still feels a bit safer for me to just return :D

Reply:
I haven't tested this, but i believe if you exit there it won't print that any tests were skipped. So i would leave it like it is.

Reply:
Okay I am fine with how you had it!